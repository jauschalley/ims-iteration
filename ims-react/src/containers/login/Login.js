import React from 'react';
import {Col, Container, Row, Form, FormLabel, Button} from 'react-bootstrap';
import {observer} from 'mobx-react';
import {Formik, Field} from 'formik';

const Login = (props) => {
  return(
    <Container fluid className="login text-align-left">
      <Col md={4}>
        <h1>Login</h1>
        <Formik
          initialValues={{email: '', password: ''}}
          onSubmit={(values) => props.authStore.login(values.email, values.password, () => window.location = '/home')}
        >
          {
            formikProps => 
              <Form>
                <Row>
                  <FormLabel>Email:</FormLabel>
                  <Field id="email" name="email" type="email" required/>
                </Row>
                <Row>
                  <FormLabel>Password:</FormLabel>
                  <Field id="password" name="password" type="password" required/>
                </Row>
                <Row>
                  <Button onClick={formikProps.handleSubmit}>Login</Button>
                </Row>
              </Form>
          }
        </Formik>
      </Col>
    </Container>
  );
};

export default observer(Login);