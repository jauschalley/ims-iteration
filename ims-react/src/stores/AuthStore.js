import {observable, action, decorate} from "mobx";
import axios from 'axios';
import config from '../config';

class AuthStore {
  constructor() {
    this.buildFromSession();
  }

  loggedIn = false;
  
  login = (username, password, successCallback) => {
    let headers = {
      "headers": {
        "credentials": window.btoa(username + ':' + password)
      }
    };
    axios.post(`${config.backendUrl}/login`, null, headers).then((response) => {
      console.log(response.headers);
      this.loggedIn = true;
      this.writeToSession(response.headers.jwt);
      successCallback();
    }).catch((err) => {
      console.log(err);
    });
  }

  logout = (successCallback) => {
    console.log('Logged out!');
    this.loggedIn = false;
    sessionStorage.clear();
    successCallback();
    //need backend logout method
    // RequestService.delete(`${config.backendUrl}/logout`, 
    //   (response) => {
    //     console.log('Logged out!')
    //     this.loggedIn = false;
    //     sessionStorage.clear();
    //     successCallback();
    //   }
    // )
  }

  writeToSession = (token) => {
    let json = JSON.stringify(this);
    sessionStorage.setItem('AuthStore', json);
    token && sessionStorage.setItem('Token', token);
  }

  buildFromSession = () => {
    let json = JSON.parse(sessionStorage.getItem('AuthStore'));
    if(json) {
      this.loggedIn = json.loggedIn;
    }
  }
}

decorate(AuthStore, {
  loggedIn: observable,
  login: action
});

export default AuthStore;