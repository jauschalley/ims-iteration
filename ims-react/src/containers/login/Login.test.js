import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';
import AuthStore from '../../stores/AuthStore';

describe('<Login />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Login authStore={new AuthStore()} />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});