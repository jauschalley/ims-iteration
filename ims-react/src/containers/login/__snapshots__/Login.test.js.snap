// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<div
  className="login text-align-left container-fluid"
>
  <div
    className="col-md-4"
  >
    <h1>
      Login
    </h1>
    <form
      className=""
    >
      <div
        className="row"
      >
        <label
          className="form-label"
        >
          Email:
        </label>
        <input
          id="email"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <label
          className="form-label"
        >
          Password:
        </label>
        <input
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="row"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</div>
`;
